### PR TITLE
Disallow non-positive spend amounts

### LINKED ISSUE #431: [1889] It's possible to buy a train from another corporation for 0Y
In game 973 one player managed to buy a train from another corporation for 0Y which should be forbidden. The move was undo.

### PR DESCRIPTION
**Pull request by sparr**
Resolves #431

### CODE REVIEW

File: lib/engine/round/operating.rb [hunk 1]
```diff
@@ -433,7 +433,7 @@ def buy_company(company, price)
         @just_sold_company = company
 
         @current_entity.companies << company
-        @current_entity.spend(price, entity)
+        @current_entity.spend(price, entity) if price.positive?
```
Comment (tobymao) [inline lib/engine/round/operating.rb:436]: this one shouldn't be here, i'll fix it

File: lib/engine/round/operating.rb [hunk 2]
```diff
@@ -492,7 +492,7 @@ def liquidate(player)
           sell_shares(bundle)
         end
 
-        player.spend(player.cash, @bank)
+        player.spend(player.cash, @bank) if player.cash.positive?
```
Comment (tobymao) [inline lib/engine/round/operating.rb:495]: this is incorrect as well

File: lib/engine/round/operating.rb (hunk 1 above)
Comment (sparr) [reply lib/engine/round/operating.rb:436]: $0 is not a valid price in 1889, but in a game where it is valid, this check will be needed.

File: lib/engine/round/operating.rb (hunk 2 above)
Comment (sparr) [reply lib/engine/round/operating.rb:495]: This one is needed for when a player goes bankrupt with zero dollars

**Pull request merged**